### ISSUE TITLE
In mixed JS/TS codebase no way to catch await without async in JS

### ISSUE THREAD

**Issue by tupelo-schneck** [typescript-eslint/typescript-eslint#2126]
Using `"parser": "@typescript-eslint/parser"` in a mixed JS/TS codebase, the linter
does not catch the issue of await in a non-async function in JS files.  (Also TS files, but the TS compiler catches the error there.)

This means that such errors get no warning or error mark in IDEs.  Using `"parser": "espree"` this gets a parsing error.

The linter would be more useful in mixed JS/TS codebases if this error could be marked.  Either the parser could complain, or a rule could be added to the recommended rules, so that linting behavior is as useful as the default ESLint parser for this case.

**Repro**

```JSON
{
    "env": {
        "browser": true,
        "node": true,
        "es6": true
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "project": "tsconfig.json",
        "ecmaVersion": 2019
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/eslint-recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:@typescript-eslint/recommended-requiring-type-checking"
    ],
    "rules": {
        "@typescript-eslint/explicit-function-return-type": "off",
        "@typescript-eslint/no-unused-vars": "off"
    }
}
```

```JS
function foo() {
    await Promise.resolve();
}
```

**Expected Result**

I expect the linter to flag the use of await without async as an error.

**Actual Result**

The file lints without errors or warnings.

**Additional Info**

Adding `"checkJs": true` to `tsconfig.json` would get tsc catching these errors in JS files, but there are so many other spurious issues raised with `"checkJs": true` that it's basically unusable.  Given that other ESLint parsers catch this error, it seems appropriate for the linter to catch this.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `3.0.2` |
| `TypeScript`                | `3.9.3` |
| `ESLint`                    | `7.1.0` |
| `node`                      | `13.5.0` |
| `npm`                       | `6.13.4` |

**Comment by bradzacher** [comment]
How are you compiling your code? Won't your build tooling catch this?
I'm curious as to why you want to rely on the linter to validate that you're writing valid JS code.

> other spurious issues

Could you give an example? I was under the impression that `checkJs` should only cause you problems if your code has problems.

**Comment by tupelo-schneck** [comment]
Compiler is tsc.  (Target is Electron so no need to further compile most JavaScript.)

If I change the source to

```JS
function foo(bar) {
    document.getElementById('textInput').value = 'hello';
    await Promise.resolve();
}
```

here are the tsc errors with `checkJs`

```
src/main.js:1:14 - error TS7006: Parameter 'bar' implicitly has an 'any' type.

1 function foo(bar) {
               ~~~

src/main.js:2:5 - error TS2531: Object is possibly 'null'.

2     document.getElementById('textInput').value = 'hello';
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/main.js:2:42 - error TS2339: Property 'value' does not exist on type 'HTMLElement'.

2     document.getElementById('textInput').value = 'hello';
                                           ~~~~~

src/main.js:3:5 - error TS1308: 'await' expressions are only allowed within async functions and at the top levels of modules.

3     await Promise.resolve();
      ~~~~~

  src/main.js:1:10
    1 function foo(bar) {
               ~~~
    Did you mean to mark this function as 'async'?
```

Some of those errors are difficult or impossible to fix in JavaScript.

If I instead use Babel to compile my TypeScript, I do get a compilation error from Babel for the await.  But that doesn't help with VS Code, which shows `tsc` errors and ESLint errors and warnings in any case.  So far, my only workaround is to have a convoluted .eslintrc.json where `"parser": "@typescript-eslint/parser"` is used only for .ts files.
